### PR TITLE
Change ConnectList representation in config [ECR-1760]

### PR DESCRIPTION
**Pull request by pavel-mukhanov**
Change `ConnectList` representation in config from:
```toml
[connect_list.peers]
16ef83ca4b231404daec6d07b24beb84d89c25944285d2e32a2dcf8f0f3eda72 = "127.0.0.1:6333"
648e98a2405a40325d946bf8de6937795fe5c22ab095bca765a8b218e49ff5a3 = "127.0.0.1:6334"
```
to:
```toml
[[connect_list.peers]]
address = "127.0.0.1:6333"
public_key = "16ef83ca4b231404daec6d07b24beb84d89c25944285d2e32a2dcf8f0f3eda72"

[[connect_list.peers]]
address = "127.0.0.1:6334"
public_key = "648e98a2405a40325d946bf8de6937795fe5c22ab095bca765a8b218e49ff5a3"
```
This was to done to simplify parsing.

### CODE REVIEW

File: exonum/src/node/mod.rs [hunk 1]
```diff
@@ -328,6 +330,46 @@ impl NodeRole {
...
+
+        ConnectListConfig { peers }
+    }
+
+    /// Creates `ConnectList` from validators keys and corresponding IP addresses.
+    pub fn from_validator_keys(validators_keys: &[ValidatorKeys], peers: &[SocketAddr]) -> Self {
+        let peers: Vec<_> = peers
+            .iter()
+            .zip(validators_keys.iter())
+            .map(|(p, v)| ConnectInfo {
```
Comment (popzxc) [inline exonum/src/node/mod.rs]: Why address is represented as `p`?

Comment (pavel-mukhanov) [reply exonum/src/node/mod.rs]: Because it comes from `peers`.

File: exonum/src/node/mod.rs [hunk 2]
```diff
@@ -328,6 +330,46 @@ impl NodeRole {
...
+/// ConnectList representation in node's config file.
+pub struct ConnectListConfig {
+    /// Peers to which we can connect.
+    pub peers: Vec<ConnectInfo>,
+}
+
+impl ConnectListConfig {
+    /// Creates `ConnectList` from validators public configs.
+    pub fn from_node_config(list: &[NodePublicConfig]) -> Self {
+        let peers: Vec<_> = list.iter()
```
Comment (ozkriff) [inline exonum/src/node/mod.rs]: Is this `: Vec<_>` really required? Rust should be able to infer a type from its direct use as a Vec field in the next statement.

File: exonum/src/helpers/fabric/details.rs [hunk 3]
```diff
@@ -34,7 +34,8 @@ use api::backends::actix::AllowOrigin;
 use blockchain::{config::ValidatorKeys, GenesisConfig};
 use crypto;
 use helpers::{config::ConfigFile, generate_testnet_config};
-use node::{ConnectList, NodeApiConfig, NodeConfig};
+use node::ConnectListConfig;
+use node::{NodeApiConfig, NodeConfig};
```
Comment (DarkEld3r) [inline exonum/src/helpers/fabric/details.rs]: Please merge these two lines. :upside_down_face:

File: exonum/src/node/mod.rs [hunk 4]
```diff
@@ -38,14 +38,15 @@ use std::{
     time::{Duration, SystemTime},
 };
 
+use blockchain::ValidatorKeys;
```
Comment (DarkEld3r) [inline exonum/src/node/mod.rs]: Same here.

File: exonum/src/node/mod.rs (hunk 2 above)
Comment (DarkEld3r) [reply exonum/src/node/mod.rs]: Type isn't needed here, right?

**Pull request merged**